ultra/ultra/baselines/agent_spec.py

```diff
@@ -79,23 +81,18 @@ def __new__(
 
                 spec = new_spec
         else:
-            base_dir = os.path.join(os.path.dirname(__file__), "../")
-            pool_path = os.path.join(base_dir, "agent_pool.json")
-
-            policy_class_name = policy_class.__name__
-            agent_name = None
-
-            with open(pool_path, "r") as f:
-                data = json.load(f)
-                agents = data["agents"].keys()
-                for agent in agents:
-                    if data["agents"][agent]["class"] == policy_class_name:
-                        agent_name = data["agents"][agent]["name"]
-                        break
-
-            assert agent_name != None
+            # If policy_params is None, then there must be a params.yaml file in the same
+            # directory as the policy_class module.
+            if not policy_params:
+                policy_class_module_file = inspect.getfile(policy_class)
+                policy_class_module_directory = os.path.dirname(
+                    policy_class_module_file
+                )
+                policy_params = load_yaml(
+                    os.path.join(policy_class_module_directory, "params.yaml")
+                )
```

Comment:
Will this implementation work with the regular train script? For static config files do we want to add a separate config path to the agent_pool?